aeppl/mixture.py

```diff
@@ -270,10 +268,6 @@ def mixture_replace(fgraph, node):
     # that belong to each one (by way of their indices).
     mixture_rvs = []
     for i, component_rv in enumerate(mixture_res):
-        if component_rv in rv_map_feature.rv_values:
-            raise ValueError(
-                f"A value variable was specified for a mixture component: {component_rv}"
-            )
```

Comment:
It doesn't make sense to raise, because this could just be an non-valued stack of valued RVs. We could check that the mixture_rv is valued here, but that seems pretty arbitrary in the context of composite logprobs. If this mixture is just a measurable input to another valued RV (and not itself valued), this would not raise, and the user would not notice something is amiss.

I am convinced the best solution is to raise explicitly in `factorized_joint_logprob`, as discussed in #90, and not in the local rewrites.